pkg/apis/submariner.io/v1/types.go

```diff
@@ -271,3 +272,61 @@ type ClusterGlobalEgressIPList struct {
 
 	Items []ClusterGlobalEgressIP `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalIngressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of desired behavior of GlobalIngressIP object.
+	Spec GlobalIngressIPSpec `json:"spec"`
+
+	// Observed status of GlobalIngressIP. Its a read-only field.
+	// +optional
+	Status GlobalIngressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalIngressIPSpec struct {
+	// TargetType can be ClusterIPService or a HeadlessServicePod
+	Target TargetType `json:"target"`
```

Comment:
I thought we concluded we don't need this field.

Reply:
I'm afraid no. We would need it as it allows us to expand the object to support regular Pods in the future.

Reply:
A regular pod would be the case where both `ServiceRef` and `PodRef` are non-nil. I thought we said that earlier.

Reply:
It is better to have a type field than infer from presence/absence of others. Maybe I missed the bit about removing this during the call. @sridhargaddam thoughts?

Reply:
I share the same opinion as @vthapar.

Reply:
ok